backend/inspirehep/accounts/views.py

```diff
@@ -88,4 +90,17 @@ def set_orcid_push_setting():
     db.session.add(orcid_account)
     db.session.commit()
 
+    if allow_push is True:
+        orcid = orcid_account.extra_data["orcid"]
+        tokens = orcid_account.remote_tokens
+
+        if len(tokens) != 1:
+            raise ValueError(
+                f"One token per remote account is expected, but found {len(tokens)} for {orcid}"
+            )
+
+        push_account_literature_to_orcid.apply_async(
+            kwargs={"orcid": orcid, "token": tokens[0].access_token}
+        )
```

Comment:
Actually I'd just get ``[0]`` and raise on IndexError instead of checking how many tokens we have.

Reply:
whole thing (and many more things) on relies on the fact that we have a single token per account, and all accounts are `orcid`, that's why there is the check.